spark/datadog_checks/spark/spark.py

```diff
@@ -642,11 +646,16 @@ def _spark_structured_streams_metrics(self, running_apps, addl_tags):
                         continue
                     groups = match.groupdict()
                     metric_name = groups['metric_name']
+                    query_name = groups.get("query_name", '')
                     if metric_name not in SPARK_STRUCTURED_STREAMING_METRICS:
                         continue
                     metric_name, submission_type = SPARK_STRUCTURED_STREAMING_METRICS[metric_name]
                     tags = ['app_name:%s' % str(app_name)]
                     tags.extend(addl_tags)
+
+                    if self._enable_query_name_tag:
+                        tags.append('query_name:%s' % str(query_name))
+
```

Comment:
```suggestion
                    if self._enable_query_name_tag:
                        query_name = groups.get("query_name", '')
                        if query_name == '':
                             tags.append('query_name:%s' % str(query_name))
                        else:
                           ... log a debug message 

```
What do you think about this?

Reply:
Now that I'm looking more carefully at the regex pattern, I don't think it will even reach this code-path if there isn't a match for `query_name` in the full metric name, due to this check:

https://github.com/DataDog/integrations-core/blob/ff8747691daf68ef04e3789ddb50fa23a961e6c3/spark/datadog_checks/spark/spark.py#L643-L646

Would it suffice if I move the debug messages to those continuation blocks instead?
```python
for gauge_name, value in iteritems(response):
    match = STRUCTURED_STREAMS_METRICS_REGEX.match(gauge_name)
    if not match:
        self.log.debug("No regex match found for gauge: '%s'" % str(gauge_name))
        continue
    groups = match.groupdict()
    metric_name = groups['metric_name']
    if metric_name not in SPARK_STRUCTURED_STREAMING_METRICS:
        self.log.debug("Unknown metric_name encountered: '%s'" % str(metric_name))
        continue
    metric_name, submission_type = SPARK_STRUCTURED_STREAMING_METRICS[metric_name]
    tags = ['app_name:%s' % str(app_name)]
    tags.extend(addl_tags)

    if self._enable_query_name_tag:
        query_name = groups['query_name']
        tags.append('query_name:%s' % str(query_name))
 
    self._set_metric(metric_name, submission_type, value, tags=tags)
```

Reply:
That looks good, thanks!

Reply:
Updated!